sdk/core/src/max_results.rs

```diff
@@ -0,0 +1,46 @@
+use crate::AppendToUrlQuery;
+use std::convert::TryFrom;
+use std::num::NonZeroU32;
+
+// This type forbids zero as value.
+// Zero is invalid in Azure and would throw an error
+// if specified.
+// Azure has a soft cap on 5k. There is no harm
+// to go over but maybe we should inform the user
+// that they are specifing a value outside the allowed range.
+// Right now we simply ignore it.
```

Comment:
Yes. About this, what do you think about `NonZeroU32`? 
It makes sense here but it's much more cumbersome to use compared to a simple `u32`. 
Should we keep it or revert to a simpler `u32` and leave to the SDK user to avoid passing values below zero?

Reply:
Hmm not sure, but being a bit more permissive might not be a bad idea if it makes using the SDK easier.